tests/Unit/NumericalAlgorithms/LinearSolver/Test_Gmres.cpp

```diff
@@ -96,6 +100,39 @@ SPECTRE_TEST_CASE("Unit.LinearSolver.Serial.Gmres",
       CHECK_ITERABLE_APPROX(second_initial_guess_in_solution_out,
                             expected_second_solution);
     }
+    {
+      INFO("Check the solver still works after serialization");
+      const auto serialized_gmres = serialize_and_deserialize(gmres);
+      DenseVector<double> second_initial_guess_in_solution_out{0., 0.};
```

Comment:
Can you define a lambda that just takes the `gmres` as an argument? You now have the same code 3x as far as I can tell